File: src/main/java/org/jukeboxmc/potion/EffectRegistry.java
package org.jukeboxmc.potion;

import java.util.HashMap;
import java.util.Map;

/**
 * @author LucGamesYT
 * @version 1.0
 */
public class EffectRegistry {

    private static final Map<EffectType, Class<? extends Effect>> EFFECT_MAP = new HashMap<>();

    public static void init() {
        register( EffectType.SPEED, SpeedEffect.class );
        register( EffectType.SLOWNESS, SlownessEffect.class );
        register( EffectType.HASTE, HasteEffect.class );
        register( EffectType.MINING_FATIGUE, MiningFatigueEffect.class );
        register( EffectType.STRENGTH, StrengthEffect.class );
        register( EffectType.INSTANT_HEALTH, InstantHealthEffect.class );
        register( EffectType.INSTANT_DAMAGE, InstantDamageEffect.class );
        register( EffectType.JUMP_BOOST, JumpBoostEffect.class );
        register( EffectType.NAUSEA, NauseaEffect.class );
        register( EffectType.REGENERATION, RegenerationEffect.class );
        register( EffectType.RESISTANCE, ResistanceEffect.class );
        register( EffectType.FIRE_RESISTANCE, FireResistanceEffect.class );
        register( EffectType.WATER_BREATHING, WaterBreathingEffect.class );
        register( EffectType.INVISIBILITY, InvisibilityEffect.class );
        register( EffectType.BLINDNESS, BlindnessEffect.class );
        register( EffectType.NIGHT_VISION, NightVisionEffect.class );
        register( EffectType.HUNGER, HungerEffect.class );
        register( EffectType.WEAKNESS, WeaknessEffect.class );
        register( EffectType.POISON, PoisonEffect.class );
        register( EffectType.WITHER, WitherEffect.class );
        register( EffectType.HEALTH_BOOST, HealthBoostEffect.class );
        register( EffectType.ABSORPTION, AbsorptionEffect.class );
        register( EffectType.SATURATION, SaturationEffect.class );
        register( EffectType.LEVITATION, LevitationEffect.class );
        register( EffectType.SLOW_FALLING, SlowFallingEffect.class );
        register( EffectType.CONDUIT_POWER, ConduitPowerEffect.class );
        register( EffectType.BAD_OMEN, BadOmenEffect.class );
        register( EffectType.HERO_OF_THE_VILLAGE, HeroOfTheVillageEffect.class );
        register( EffectType.DARKNESS, DarknessEffect.class );
        register( EffectType.FATAL_POISON, FatalPoison.class );
    }

    private static void register( EffectType effectType, Class<? extends Effect> clazz ) {
        EFFECT_MAP.put( effectType, clazz );
    }

    public static Class<? extends Effect> getEffectClass( EffectType effectType ) {
        return EFFECT_MAP.get( effectType );
    }

}

File: JukeboxMC-Server/src/main/kotlin/org/jukeboxmc/server/effect/EffectRegistry.kt
package org.jukeboxmc.server.effect

import org.jukeboxmc.api.effect.Effect
import org.jukeboxmc.api.effect.EffectType

class EffectRegistry {

    companion object {

        private var effectMap: MutableMap<EffectType, Class<out Effect>> = mutableMapOf()

        init {
            register(EffectType.SPEED, SpeedEffect::class.java)
            register(EffectType.SLOWNESS, SlownessEffect::class.java)
            register(EffectType.HASTE, HasteEffect::class.java)
            register(EffectType.MINING_FATIGUE, MiningFatigueEffect::class.java)
            register(EffectType.STRENGTH, StrengthEffect::class.java)
            register(EffectType.INSTANT_HEALTH, InstantHealthEffect::class.java)
            register(EffectType.INSTANT_DAMAGE, InstantDamageEffect::class.java)
            register(EffectType.JUMP_BOOST, JumpBoostEffect::class.java)
            register(EffectType.NAUSEA, NauseaEffect::class.java)
            register(EffectType.REGENERATION, RegenerationEffect::class.java)
            register(EffectType.RESISTANCE, ResistanceEffect::class.java)
            register(EffectType.FIRE_RESISTANCE, FireResistanceEffect::class.java)
            register(EffectType.WATER_BREATHING, WaterBreathingEffect::class.java)
            register(EffectType.INVISIBILITY, InvisibilityEffect::class.java)
            register(EffectType.BLINDNESS, BlindnessEffect::class.java)
            register(EffectType.NIGHT_VISION, NightVisionEffect::class.java)
            register(EffectType.HUNGER, HungerEffect::class.java)
            register(EffectType.WEAKNESS, WeaknessEffect::class.java)
            register(EffectType.POISON, PoisonEffect::class.java)
            register(EffectType.WITHER, WitherEffect::class.java)
            register(EffectType.HEALTH_BOOST, HealthBoostEffect::class.java)
            register(EffectType.ABSORPTION, AbsorptionEffect::class.java)
            register(EffectType.SATURATION, SaturationEffect::class.java)
            register(EffectType.LEVITATION, LevitationEffect::class.java)
            register(EffectType.SLOW_FALLING, SlowFallingEffect::class.java)
            register(EffectType.CONDUIT_POWER, ConduitPowerEffect::class.java)
            register(EffectType.BAD_OMEN, BadOmenEffect::class.java)
            register(EffectType.HERO_OF_THE_VILLAGE, HeroOfTheVillageEffect::class.java)
            register(EffectType.DARKNESS, DarknessEffect::class.java)
            register(EffectType.FATAL_POISON, FatalPoison::class.java)
        }

        private fun register(effectType: EffectType, clazz: Class<out Effect>) {
            this.effectMap[effectType] = clazz
        }

        fun getEffectClass(effectType: EffectType?): Class<out Effect>           {
            return this.effectMap[effectType]!!
        }
    }

}